modules/core/src/smithy4s/StaticService.scala

```diff
@@ -0,0 +1,37 @@
+package smithy4s
+
+import smithy4s.kinds.FunctorK5
+import smithy4s.kinds.PolyFunction5
+
+trait StaticService[P[_[_, _, _, _, _]]] extends FunctorK5[P] {
```

Comment:
After some thinking `Static` is a confusing denomination for this. I think we should call it `ServiceProduct` because it's essentially abstracting over services that are represent as a product of functions. 

@kubukoz , thoughts ?

Reply:
Also, some scaladoc would be appreciated

Reply:
agree, there's already a use of `Dynamic` in the codebase so using Static here is probably confusing. I see where the `ServiceProduct` is going to, could it be something like `ServiceWithoutInputs` ?

Reply:
> ServiceWithoutInputs 

not really. It's just a different way of passing inputs. It's similar to the difference between methods and functions.

Reply:
Okay for Record

Reply:
added some scaladocs at d8be76689ac789d83716dd1e18b81a83143af202 let me know if those are accurate 

let me know if you prefer `ServiceRecord` as the PR right now uses `ServiceProduct`

Reply:
I'm good with either